src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -102,6 +102,8 @@ function WebGLShadowMap( _renderer, _objects, maxTextureSize ) {
 			const light = lights[ i ];
 			const shadow = light.shadow;
 
+			if ( shadow.autoUpdate === false && shadow.needsUpdate === false ) continue;
+
 			if ( shadow === undefined ) {
```

Comment:
@mrdoob This is now unreachable.

The line above will instead raise an error, where a console warning would have previously been raised.

Reply:
I think this can be solved by simply move `if ( shadow.autoUpdate === false && shadow.needsUpdate === false ) continue;` under the `undefined` check.

This avoids a runtime error when e.g. users set `castShadow` to `true` for an `AmbientLight`.